pkg/controller/certificates/controller.go

```diff
@@ -121,33 +125,48 @@ func NewCertificateController(kubeClient clientset.Interface, syncPeriod time.Du
 // Run the main goroutine responsible for watching and syncing jobs.
 func (cc *CertificateController) Run(workers int, stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
+	defer cc.queue.ShutDown()
+
 	go cc.csrController.Run(stopCh)
+
+	if !framework.WaitForCacheSync(stopCh, cc.csrStoreSynced) {
```

Comment:
You don't usually have to wait for your primary caches to sync.  Consider the a replicaset controller:
 1. RS cache to get the current state of the RS
 1. Pod cache to count the number of pods already in existence for the RS

You have to be sure that the pod cache is completely filled, or the list operation to count the number of available pods won't be right.  You don't have to be sure that the RS cache is filled because once an RS is added, it will be queued and the subsequent retrieval will always work.

I don't think we have to wait on CSRs in this case because enqueuing means that our cache will have the data.   Does that make sense?